soh/src/overlays/actors/ovl_En_Ma1/z_en_ma1.c

```diff
@@ -90,7 +90,16 @@ static void* sEyeTextures[] = {
     gMalonChildEyeClosedTex,
 };
 
+bool Randomizer_ObtainedMalonHCReward() {
+    return Flags_GetEventChkInf(0x12);
+}
+
 u16 EnMa1_GetText(GlobalContext* globalCtx, Actor* thisx) {
+    // Special case for Malon Hyrule Castle Text. Placing it here at the beginning
+    // has the added benefit of circumventing mask text if wearing bunny hood.
```

Comment:
not sure how i feel about the skipping mask text stuff (feels like that should be a cvar), but since it's only in rando it's cool for now